test/index.test-d.ts

```diff
@@ -0,0 +1,8 @@
+import { expectType } from 'tsd'
+import test, { configure } from '../index.js'
+
+configure({ serial: true })
+
+test('some test', async (assert) => {
+  expectType<boolean>(await assert.is(true, true))
+})
```

Comment:
can we check the types of exports (similar to https://github.com/pinojs/pino/blob/master/test/types/pino-import.test-d.ts but within a `test` as it is already, is great), and the types on `assert` - the resolved boolean values I'm less bothered about but we can check those as well